proto-lens-benchmarks/src/Data/ProtoLens/BenchmarkUtil.hs

```diff
@@ -0,0 +1,77 @@
+-- Copyright 2016 Google Inc. All Rights Reserved.
+--
+-- Use of this source code is governed by a BSD-style
+-- license that can be found in the LICENSE file or at
+-- https://developers.google.com/open-source/licenses/bsd
+
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | A utility library for writing proto-lens benchmarks.
+module Data.ProtoLens.BenchmarkUtil (protoBenchmark, benchmarkMain) where
+
+import Control.DeepSeq (NFData)
+import Criterion.Main
+import qualified Criterion.Main.Options as Criterion
+import qualified Data.ByteString as BS (length)
+import Data.Maybe (fromMaybe)
+import Data.ProtoLens
+import Options.Applicative
+
+-- | Generate a group of benchmarks for encoding and decoding the given proto
+-- message. Includes benchmarks for decoding to both weak head normal form and
+-- normal form (i.e. all fields fully forced.) There is only one encoding
+-- benchmark because protos are encoded as strict ByteStrings, so there is no
+-- difference between normal form and weak head normal form.
+protoBenchmark
+    :: forall a.
+       (NFData a, Message a)
+    => String
+    -- ^ Benchmark name.
+    -> a
+    -- ^ Protocol buffer message to encode/decode.
+    -> Benchmark
+protoBenchmark groupName proto =
+    bgroup
+        fullGroupName
+        [ bench "encode" $ nf encodeMessage proto
```

Comment:
I wonder if it's worth trying a version of the benchmark that doesn't go through `NFData`, instead just re-encoding the messages and using the fact that `ByteString`s are trivially in WHNF?  E.g. 

    bench "encode-decoded" $ nf (fmap encodeMessage . decodeMessage') encodedProto

Let me know whether you think that would add any value.